### ISSUE TITLE
ci: snapshot process test is too flaky

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#3361]
This test is failing a lot:

- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/10386743239/attempts/1#summary-28759182888
- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/10380957419/attempts/1#summary-28742131702
- https://github.com/inspektor-gadget/inspektor-gadget/actions/runs/10393961200/attempts/1#summary-28783086720

#3338 tried to fix it but it's still failing.

**Comment by andoriyaprashant** [comment]
Hello @mauriciovasquezbernal 
I'd like to work on this issue and resolve the test failures. Could you please assign it to me
Thanks

**Comment by mauriciovasquezbernal** [comment]
@andoriyaprashant have you been able to make any progress on this?

**Comment by mauriciovasquezbernal** [comment]
It seems the snapshot_socket is flaky as well. What are we doing wrong for snapshotters?

**Comment by mauriciovasquezbernal** [comment]
I think the reason of this is that the timeout set by the client could be too low that the [message pump](https://github.com/inspektor-gadget/inspektor-gadget/blob/39da0258c782023ab7c5d36dea85bbe16907f7d0/pkg/gadget-service/service-oci.go#L184-L194) doesn't have time to send the message to the client. 

This can be reproduced by applying the following diff:

```bash 
diff --git a/pkg/gadget-service/service-oci.go b/pkg/gadget-service/service-oci.go
index 7094fa64d..59a02d198 100644
--- a/pkg/gadget-service/service-oci.go
+++ b/pkg/gadget-service/service-oci.go
@@ -184,6 +184,7 @@ func (s *Service) RunGadget(runGadget api.GadgetManager_RunGadgetServer) error {
                        go func() {
                                // Message pump to handle slow readers
                                for {
+                                       time.Sleep(2 * time.Second)
                                        select {
                                        case ev := <-outputBuffer:
                                                runGadget.Send(ev)
```

```bash 
$ go build ./cmd/gadgetctl/ && go build ./cmd/ig/

# on a different terminal 
$  sudo ig daemon


# with timeout=1 nothing is printed
$ sudo ./gadgetctl run snapshot_process:latest --timeout 1

# it works fine with timeout=3
$ sudo ./gadgetctl run snapshot_process:latest --timeout 3
RUNTIME.CONTAINERNAME                                                                             COMM                    PID        TID
laughing_bartik                                                                                   sh                    20169      20169
```

I don't think we can do other than increasing the timeout on the tests (#3760), a proper solution is to fix #3033.
